pertpy/preprocessing/_guide_rna.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+import logging
+
+import numpy as np
+import scipy
+from anndata import AnnData
+
+
+class GuideAssignment:
+    """Offers simple guide assigment based on count thresholds."""
+
+    def assign_by_threshold(
+        self,
+        adata: AnnData,
+        assignment_threshold: float,
+        layer: str | None = None,
+        output_layer: str = "assigned_guides",
+        only_return_results: bool = False,
+    ) -> np.ndarray | None:
+        """Simple threshold based gRNA assignment function.
+
+        Each cell is assigned to gRNA with at least `assignment_threshold` counts.
+
+        Args:
+            adata: Annotated data matrix containing gRNA values
+            assignment_threshold: If a gRNA is available for at least `assignment_threshold`, it will be recognized as assigned.
+            layer: Key to the layer containing raw count values of the gRNAs.
+                   adata.X is used if layer is None.
+            output_layer: Assigned guide will be saved on adata.layers[output_key]. Default value is `assigned_guides`.
+            only_return_results: If True, input AnnData is not modified and the result is returned as an np.ndarray.
+        """
+        counts = adata.X if layer is None else adata.layers[layer]
+        if scipy.sparse.issparse(counts):
+            counts = counts.A
+
+        # TODO: move to utils
+        max_entry = counts.max()
+        if not abs(max_entry - int(max_entry)) < 1e-8:
```

Comment:
IMO I'd add to the docs of this that this function expects unnormalized data. It doesn't make sense to use this on normalized data. Then remove this.